javascript/javascript2/week2/hyfBay/main.js

```diff
@@ -0,0 +1,131 @@
+(function () {
+  ("use strict");
+  const products = getAvailableProducts();
+  const productsUl = document.querySelector("section.products ul");
+
+  function renderProducts(products) {
+    products.forEach((product) => {
+      const li = document.createElement("li");
+
+      let shipsToHTML = "";
+      product.shipsTo.forEach(
+        (country) => (shipsToHTML += `<li>${country}</li>`)
+      );
+
+      li.innerHTML = `
+            <ul class="products-features" data-price=${product.price}>
+                <li class="product-name">${product.name}</li>
+                <li class="product-price">${product.price}</li>
+                <li class="product-rating">${product.rating}</li>
+                <ul class="ships-to country">${shipsToHTML}</ul>
+            </ul>
+        `;
+      productsUl.appendChild(li);
+    });
+  }
+
+  renderProducts(products);
+  // variables for getting elements
+  const searchInput = document.querySelector("#searchName");
+  const maxPrice = document.querySelector("#maxPrice");
+  const listItemsUl = document.querySelectorAll(".products-features");
+
+  // ################ 2.1. Filter products
+
+  const filterElements = document.querySelector(".country select");
+
+  filterElements.addEventListener("change", () => {
```

Comment:
You can also access the values of the different option elements by passing an event parameter to the eventListener function, and then accessing event.target.value